### ISSUE TITLE
[css-images-4] Restriction on `<percentage>` values in `stripes()`

### ISSUE THREAD

**Issue by cdoublev** [w3c/csswg-drafts#9551]
`<percentage>` is currently restricted to `[0,100]` in [`stripes()`](https://drafts.csswg.org/css-images-4/#funcdef-stripes) otherwise the function is invalid:

  > `<color-stripe> = <color> && [ <length-percentage> | <flex> ]?`
  >
  > - **`<percentage [0,100]>`**: [...] Only values between `0%` and `100%` (inclusive) are valid.
  > - **`<length [0,∞]>`** : [...] Negative length values are invalid.

I agree with https://github.com/w3c/csswg-drafts/pull/9516#discussion_r1379437798 that values outside this range are meaningless.

But then I am not sure why they make sense in `<alpha-value>` or `<brightness()>` (for example), which produce different animating behaviors (because `<alpha-value>` is clamped at parse time).

However my main concern is `<length-percentage>` expanding to `<length [0,∞]> | <percentage [0,100]>`, which is not great for grammar driven parser, therefore I suggest this change:

```diff
- <color-stripe> = <color> && [ <length-percentage> | <flex> ]?
+ <color-stripe> = <color> && [ <length [0,∞]> | <percentage [0,100]> | <flex> ]?
```

Note that `<length-percentage [0,]>` is not problematic because `[0,]` can be "inherited" during expansion.

**Comment by Loirooriol** [comment]
Yeah, I don't think it makes much sense to forbid `101%` but allow `calc(1px / 0)`.

And as you say the grammar uses `<length-percentage>`, it's not clear how the restriction applies to e.g. `calc(101% - 100px)`.

But rather than dropping `<length-percentage>`, I would just drop the restriction and define it correctly.

**Comment by cdoublev** [comment]
I do not think `calc(-1% + 1%)` should be an invalid `<percentage [0,100]>` and:

  > Parse-time range-checking of values is not performed within math functions, and therefore out-of-range values do not cause the declaration to become invalid.

https://drafts.csswg.org/css-values-4/#calc-range

But I wonder why some values are invalid when out of range, instead of being consistently clamped at computed value time, so I am also fine with dropping the restriction. Cc @SebastianZ, who may disagree.

**Comment by SebastianZ** [comment]
Clamping the values instead of disallowing them seems fine to me. Question is whether it should happen at computed value time as suggested by @cdoublev or already at parsed value time to be consistent with [`<alpha-value>`](https://drafts.csswg.org/css-color-4/#alpha-syntax).

Sebastian

**Comment by Loirooriol** [comment]
I don't think this needs to be consistent with `<alpha-value>`, which is a pure percentage, here percentages resolve against a length and have type `«[ "length" → 1 ]»` so it's an entirely different situation.

So I would just rely on

> The total width is defined by the context in which the [stripes()](https://drafts.csswg.org/css-images-4/#funcdef-stripes) function is used. If the sum of the stripes is smaller than the total width, the [paint line](https://drafts.csswg.org/css-images-4/#paint-line) is [transparent black](https://www.w3.org/TR/css-color-4/#transparent-black) for its remaining length, as if a final [transparent](https://www.w3.org/TR/css-color-4/#valdef-color-transparent) argument were given. **If the sum is larger, any stripes or portions beyond the total width are truncated.**

I don't see the need to truncate individual values at all.
